### PR TITLE
Added details about support for domains

### PR DESCRIPTION
**Pull request by DandelionSprout**
As support for plain domains rules was added to AdGuard Home a couple months ago, I've added details about that to your help docs.

I also added two examples that I felt could be informational (Especially №8, to show how it can do something that few other adblock tools can do), and corrected some grammar errors.

### CODE REVIEW

File: 02.general/19.dns-filtering-syntax/docs.en.md [hunk 1]
```diff
@@ -18,14 +18,21 @@ DNS filtering supports two types of rules:
...
 Examples:
 
-1. `||example.org^`- block access to the example.org domain and all it's subdomains
-2. `@@||example.org^` - unblock access to the example.org domain and all it's subdomains
-3. `0.0.0.0 example.org` - AdGuard will now return 0.0.0.0 address for the example.org domain (but not it's subdomains)
+1. `||example.org^`- block access to the example.org domain and all its subdomains
+2. `@@||example.org^` - unblock access to the example.org domain and all its subdomains
+3. `0.0.0.0 example.org` - AdGuard will now return 0.0.0.0 address for the example.org domain (but not its subdomains)
+4. `example.org` - treated as equivalent to #3
 4. `! Here goes a comment` - just a comment
```
Comment (szolin) [inline 02.general/19.dns-filtering-syntax/docs.en.md:29]: `4.` again

File: 02.general/19.dns-filtering-syntax/docs.en.md [hunk 2]
```diff
@@ -18,14 +18,21 @@ DNS filtering supports two types of rules:
     * Rules with other modifiers will be ignored
 
 * “Hosts” rules, which is basically the same as `/etc/hosts`
+* Domains rules, which are simply domains
```
Comment (szolin) [inline 02.general/19.dns-filtering-syntax/docs.en.md:21]: I'd rather say just `Domain names`.  `Domains rules` looks strange to me.

File: 02.general/19.dns-filtering-syntax/docs.en.md [hunk 3]
```diff
@@ -18,14 +18,21 @@ DNS filtering supports two types of rules:
...
+2. `@@||example.org^` - unblock access to the example.org domain and all its subdomains
+3. `0.0.0.0 example.org` - AdGuard will now return 0.0.0.0 address for the example.org domain (but not its subdomains)
+4. `example.org` - treated as equivalent to #3
 4. `! Here goes a comment` - just a comment
 5. `# Also a comment` - just a comment
 6. `/REGEX/` - block access to the domains matching the specified regular expression
 
-Use this syntax to create or import rules to DNS filter. If you have any questions, feel free to post them on our [forum](https://forum.adguard.com/index.php?forums/69/).
\ No newline at end of file
+Advanced examples:
```
Comment (szolin) [inline 02.general/19.dns-filtering-syntax/docs.en.md:33]: Why are the next 2 examples are "advanced", while `/REGEX/` example isn't?
I don't think we should split them into 2 groups.

File: 02.general/19.dns-filtering-syntax/docs.en.md [hunk 4]
```diff
@@ -18,14 +18,21 @@ DNS filtering supports two types of rules:
...
+
+7. `||example.*^` - wildcard asterisks can be used in domains (Also works with simply `example.*`)
+8. `93.184.216.34 example.org` - AdGuard is also able to return non-localhost addresses for domains
```
Comment (szolin) [inline 02.general/19.dns-filtering-syntax/docs.en.md:36]: The point is not just to `return non-localhost addresses` but rather any arbitrary IP address which the browser will be redirected to (or which DNS client will get in response).

**Review by szolin — CHANGES_REQUESTED** [review]

**Comment by DandelionSprout** [comment]
This PR was made back when support for things like raw domains and wildcarded domains in AdGuard DNS were considered brand-new and revolutionary, which it presumably isn't anymore. Plus, in the 54 weeks that followed the PR, I've also realised that `||example.*^` should have been written as `||example.`.

So I acknowledge that I'd actually have to rewrite this PR almost from scratch. Give me 2 hours.

**Comment by DandelionSprout** [comment]
Okay, let's see if it looks any better to you guys now.

File: 02.general/19.dns-filtering-syntax/docs.en.md [hunk 5]
```diff
@@ -18,14 +18,21 @@ DNS filtering supports two types of rules:
...
+
+Use this syntax to create or import rules to a DNS filter. If you have any questions, feel free to post them on our [forum](https://forum.adguard.com/index.php?forums/69/).
```
Comment (el-termikael) [inline 02.general/19.dns-filtering-syntax/docs.en.md]: Based on the examples above, you can create rules for DNS filters. If you have any questions, feel free to post them on our [forum](https://forum.adguard.com/index.php?forums/69/).

Comment (DandelionSprout) [reply 02.general/19.dns-filtering-syntax/docs.en.md]: Thanks for the suggestion. I seem to have implemented it now.

**Review by el-termikael — DISMISSED** [review]

**Comment by slavaleleka** [comment]
@DandelionSprout @el-termikael  
I've fixed the conflict. could you please check one more time?

**Comment by DandelionSprout** [comment]
Looks good enough to me; as although AdGuard Home has added *many* new syntaxes in the past months (e.g. `$client`), none of them were also added to AdGuard for Android's DNS filtering.

**Review by el-termikael — APPROVED** [review]

**Pull request merged**